database/merkledb/batcher.go

```diff
@@ -0,0 +1,96 @@
+package merkledb
+
+import (
+	"github.com/ava-labs/avalanchego/database"
+)
+
+type batchValue struct {
+	key    []byte
+	value  []byte
+	delete bool
+}
+
+// Batch is a write-only database that commits changes to its host database
+// when Write is called. A batch cannot be used concurrently.
+type Batch struct {
+	data []batchValue
+	tree *Tree
+}
+
+func NewBatch(t *Tree) database.Batch {
+	return &Batch{
+		data: []batchValue{},
+		tree: t,
+	}
+}
+
+// ValueSize retrieves the amount of data queued up for writing.
+func (b *Batch) ValueSize() int {
+	return len(b.data)
+}
+
+// Write flushes any accumulated data to disk.
+func (b *Batch) Write() error {
+
+	var err error
+
+	for _, d := range b.data {
```

Comment:
Can you elaborate ?
I think we're already writing in memory and commiting at the end.

No problem adding the TODO - curious on what the direction :)

Reply:
For example, if a batch contains two insertions, we will update all the parents these insertions share twice. (Albeit in memory). If the size of the batch is smaller than the trie, which it will be in all realistic cases, then it makes sense to do some pre-processing such that the batched operations don't modify the same part of the trie multiple times.

Reply:
Done :)